BoardPagecontroller 관련 service,entity logic,naming변경 (#110)

jeeseungbae opened the pull request.

Commit:
refector : boardFormPage 관련 service,entity logic,naming변경

Method create modify

Commit:
refector : boardFormPage 관련 service,entity logic,naming변경

Method create modify

Commit:
fix : @Param 어노테이션 제거 application 실행 오류

Commit:
refector : refector : boardFormPageController 관련 service,entity logic,naming변경

CategoryLogicService Method 추가 및 테스트 진행
CategoryRepository Method 추가 및 테스트 진행

Commit:
refector : refector : boardFormPageController 관련 service,entity logic,naming변경

BoardLogicService -> Method naming 및 logic 변경

Commit:
refector : refector : boardFormPageController 관련 service,entity logic,naming변경

jeeseungbae committed:
refector : BoardLogicService 디미터 법칙적용 Method logic 변경

Commit:
Merge branch 'feat/112' of https://github.com/jeeseungbae/teamproject_final into feat/112

Commit:
refector : BoardLogicService,AnswerLogicService클래스 생성자로 의존성을 주입받는 경우로 변경

Commit:
style : 자동포멧팅 기능 설정

Commit:
refector : Systemprintln 사용 지양 logic변경

Commit:
refector : BoardFormPageController 데이터 주입 logic 변경

jeeseungbae committed:
Merge branch 'master' into feat/112

Commit:
Merge branch 'master' of https://github.com/playdata-finalproject/teamproject_final into feat/112

Commit:
refector : BoardFormPageController 데이터 주입 logic 변경

Commit:
refector : BoardPageController logic 및 namging 변경

Commit:
refector : BoardPagecontroller 관련 service,entity logic,naming변경

BoardLogicService : findCategoryIdTitleContents 메소드 분리 로직 변경
SearchForm 클래스 추가 후 if문 없이 검색허용 빈 주입
BoardLogicServiceTest : 수정후 오류값 존재 -> 제거

Commit:
refector : SearchForm HashMap 초기화 및 naming 변경

hyojaekim commented on app/src/main/java/com/finalproject/shelter/business/service/logic/BoardLogicService.java:
```diff
@@ -43,23 +43,13 @@
-        return null;
+    public Page<Board> findTitle(String id,String searchText, Pageable pageable){
+        return boardRepository.findBoardByCategoryIdAndTitleContainingAndContentsContainingOrderByRegisteredAtDescIdDesc
+                (Long.parseLong(id),searchText,"",pageable);
```
빈 문자열을 넣는 이유가 있을까요??!
빈 문자열을 넣고 있다는 것은 잘못된 설계가 아닌지 의심해보면 좋을 것 같아요ㅎㅎ

그리고 쿼리 메서드가 현재는 상당히 길다는 생각이 드네요
메서드 명은 간결하게 하고, 쿼리를 직접 작성하는 것은 어떨까요?! :)

jeeseungbae replied:
네 빈 문자열을 넣는다면 전체 게시판페이지를 볼수 있게 해놨는데 사실상 필요가 없을수도 있네요ㅎㅎ 쿼리를 나누거나 직접 작성할 예정이지만 Controller 부터 차례로 고칠 예정이라 시간이 좀 걸릴 것 같습니다!!!ㅎㅎㅎ

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -2,54 +2,51 @@
-    @Autowired
-    private BoardLogicService boardLogicService;
+    private final BoardLogicService boardLogicService;
+    private final SearchForm searchForm;
```
final은 어떤 이유로 사용하셨나요?!

jeeseungbae replied:
Controller 내부에서 변경이 가능하지 못하게 사용하였습니다. 
주입받은 service를 일관성 있게 제공하기 위해서입니다.

hyojaekim replied:
좋아용!
final이 클래스, 메서드, 필드에 붙으면 각각 어떤 차이가 있는지 알아보면 좋을 것 같네요! ㅎㅎ

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -2,54 +2,51 @@
-        modelAdds(boardLogicService.bestmonthview(id),model,"mothview");
+        model.addAttribute("boards", boards);
+        model.addAttribute("eachboard", boardLogicService.readCategory(id));
+        model.addAttribute("startPage", Math.max(1, boards.getPageable().getPageNumber() - 4));
```
매직 넘버에 대해서 알아보면 좋을 것 같아요!

해당 숫자는 어떤 의미를 가지는지 파악하기가 쉽지 않아요!
매직 넘버에 대해서 알아보고, 어떻게 리팩토링 할 지 고민해보면 좋을 것 같네용 👍

jeeseungbae replied:
감사합니다!! 참조하겠습니다ㅎㅎㅎㅎㅎㅎ

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -2,54 +2,51 @@
+        model.addAttribute("boards", boards);
+        model.addAttribute("eachboard", boardLogicService.readCategory(id));
+        model.addAttribute("startPage", Math.max(1, boards.getPageable().getPageNumber() - 4));
+        model.addAttribute("endPage", Math.min(boards.getTotalPages(), boards.getPageable().getPageNumber() + 4));
```
여기도 마찬가지에요! :)

jeeseungbae replied:
넵 👍

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -2,54 +2,51 @@
-    public String deleteboard(@RequestParam(value = "id") String ids) {
-        boardLogicService.deleteid(ids);
+    public String delete(@RequestParam(value = "id") String id) {
+        boardLogicService.deleteid(id);
```
👍

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -2,54 +2,51 @@
         return "pages/list";
     }
 
     @GetMapping("/delete")
```
동사를 HTTP 메서드로 표현하면 좋을 것 같아요! ㅎㅎ

아직 리팩토링 하지 않은 부분이긴 한데
REST API에 대해서 알아보고, 적용할 점이 없는지 고민해보면 좋을 것 같네용 :)

예전에 제가 좋게 봤던 영상인데 아래 링크 참고하시면 괜찮을 것 같아요 ㅎㅎ
https://tv.naver.com/v/2292653

jeeseungbae replied:
좋은정보 감사합니다!!ㅎㅎㅎ

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/settingform/SearchForm.java:
```diff
@@ -0,0 +1,29 @@
+
+@Component
+public class SearchForm {
+    private HashMap<String, Page<Board>> list;
```
구체적인 구현체보다 추상화되어 있는 인터페이스를 사용하는 것은 어떨까요?

ex) HashMap -> Map
ex) ArrayList -> List

jeeseungbae replied:
객체가 모호해질 수 있다고 생각이 들어 구체적인 구현체를 사용하였는데 사용 용도에 따라 추상화 되어있는 인터페이스도 고려해봐야 할 것 같습니다. 감사합니다~~

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/settingform/SearchForm.java:
```diff
@@ -0,0 +1,29 @@
+
+    @PostConstruct
+    public void init(){
+        list = new HashMap<>();
```
생성자에서 초기화하지 않고, 해당 메서드에서 초기화한 이유가 있을까요?!

jeeseungbae replied:
이유는 없습니다. 여러가지 방법을 사용하고 싶었는데 @PostConstruct로 초기화하는 방법이 있어서 사용 해봤습니다.

jeeseungbae replied:
@Postconstruct로 초기화하는 방식이 있어서 사용해보았습니다. 현재는 의미가 없지만 setter메서드를 사용하게 된다면 종속성이 있는 경우 생성자가 개체를 완전히 초기화 할 수 없다고 공부하여 조금씩 사용해보려고 초기화하였습니다.

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/settingform/SearchForm.java:
```diff
@@ -0,0 +1,29 @@
+
+@Component
+public class SearchForm {
+    private HashMap<String, Page<Board>> list;
```
list라는 이름은 적절하지 않아보여요ㅜ
변수 이름에 자료형은 사용하지 않고, 네이밍하는 것은 어떨까요?

https://woowacourse.github.io/javable/post/2020-04-24-variable_naming/

jeeseungbae replied:
아 맞네요 list라는 이름을 임의로 사용하였는데 고치지 못하였습니다. 감사합니다!!

hyojaekim reviewed (commented):
점점 좋아지고 있어요! 👍 👍 

피드백은 정답은 없으니 모두 반영하지 않아도 돼요!
반영하면서도 다른 방안이 있다면 코멘트 남겨주세요! ㅎㅎ

(자동 포맷팅 기능을 활용해보세요! ㅎㅎ)

Commit:
refector : BoardPagecontroller 관련 service,entity logic,naming변경

BoardLogicService Method 추가 - 기본 카테고리 클릭시 페이지 전달
BoardRepository Method 추가 - 게시판 목록 전달
BoardPageController - 매직넘버 적용, 페이지 검색, 카테고리 클릭 메서드 분할
SearchForm - 빈 문자열 삭제

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/controller/view/board/BoardPageController.java:
```diff
@@ -26,17 +28,34 @@ public BoardPageController(BoardLogicService boardLogicService, SearchForm searc
+        model.addAttribute("startPage", Math.max(pageMinNumber, boards.getPageable().getPageNumber() - pageRange));
+        model.addAttribute("endPage", Math.min(boards.getTotalPages(), boards.getPageable().getPageNumber() + pageRange));
+        model.addAttribute("weekview", boardLogicService.bestweekview(id));
+        model.addAttribute("mothview", boardLogicService.bestmonthview(id));
```
카멜 케이스를 따르면 좋을 것 같아요! ㅎ

jeeseungbae replied:
미처 확인하지 못햇네요 감사합니다!

hyojaekim commented on app/src/main/java/com/finalproject/shelter/presentation/settingform/SearchForm.java:
```diff
@@ -12,18 +11,19 @@
+
+    public void add(BoardLogicService boardLogicService, String id, String searchText, Pageable pageable) {
+        boards.put("title", boardLogicService.findTitle(id, searchText, pageable));
+        boards.put("contents", boardLogicService.findContents(id, searchText, pageable));
```
BoardLogicService를 넘겨주는 것보다 외부에서 찾은 타이틀이나 컨텐츠를 넘겨주는 것은 어떨까요?

hyojaekim replied:
코드는 짧지만 이 부분에서 여러가지 고민을 할 수 있을 것 같아요! ㅎ

1. 해당 메서드는 몇 가지의 일을 하고 있을까요?!
2. 서비스를 넘겨주는 것보다 인자로 넘겨줄 때 필요한 데이터를 미리 찾아서 넘겨주는 것이 어떨까요?
3. 파라미터의 개수가 많으면 어떤 문제가 있을 수 있을까요?

이부분에 대해서 고민해보면 좋을 것 같네용 😄

jeeseungbae replied:
이해가 되지 않았습니다. 
1. boardLogicService.findTilte()은 반환객체가 Page<Board>인데 영향이 있는건가요??? 
2.HashMap에 Page<Board>를 넣어주는데 미리 찾아서 넣어주는게 아닌건가요???

파라미터의 개수를 신경쓰지 못한것 같습니다. 
좀 더 고민해보겠습니다!! 감사합니다~~👍🏻👍🏻👍🏻

hyojaekim reviewed (approved):
좋아요~~ 잘했어용 👍🏻

Commit:
boardLogicService Method 카멜케이스 적용

Commit:
refector : Map 인터페이스 적용

Commit:
refector : SearchData 클래스 생성 SearchData 파라미터 값 연결

Commit:
refector : SearchForm 클래스 SearchData 파라미터 값 연결

Commit:
style : 안쓰는 변수 제거 👊

hyojaekim reviewed (approved):
리팩토링 잘하셨습니다! 👍🏻
궁금하신 부분들은 따로 연락해서 말씀드리도록 할게요! ㅎ
고생하셨습니당

Commit:
style : 변수명 변경
